src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -57,7 +57,7 @@ public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AppTokenPro
             }
         }
 
-        protected virtual async Task<ManagedIdentityResponse> HandleResponseAsync(
+        protected virtual Task<ManagedIdentityResponse> HandleResponseAsync(
```

Comment:
There is no `await` in this method.

Reply:
Right, it is in one of the overridden methods.

Reply:
That's ok. You can omit `async` in the the class hierarchy, i.e. `async` is not part of the contract. Only the return type `Task<foo>` is important.